Controllers/window_controller.py

```diff
@@ -617,19 +743,24 @@ def dynamic_button_click(self, button_definition):
         Parameters:
             button_definition - An instance of ButtonDefinition
         """
-        video_timestamp = self._window.media_panel.media_control_panel.time_stamp.text()
-        split_one = video_timestamp.split("Time: ")
-        split_two = split_one[1].split("/")
-        video_timestamp = split_two[0]
-        for row in range(self._window.table_panel.table.rowCount()):
-            column = 0
-            cell = self._window.table_panel.table.item(row, column)
-            if not cell:
-                timestamp_text = QTableWidgetItem(video_timestamp)
-                self._window.table_panel.table.setItem(row, column, timestamp_text)
-                column = 1
-                while column < self._window.table_panel.table.columnCount():
-                    insert_text = QTableWidgetItem(button_definition.data[column])
-                    self._window.table_panel.table.setItem(row, column, insert_text)
-                    column += 1
-                return
+        if self._media_player.hasVideo():
```

Comment:
Instead of wrapping the logic of the entire method in this if-statement, we can change it to the following:

`
if not self._media_player.hasVideo():
    return
# rest of function
`